File: tests/src/org.scala-lang/scala-reflect/2.11.8/src/test/scala/org_scala_lang/scala_reflect/ReflectionUtilsTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_scala_lang.scala_reflect

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

import scala.reflect.internal.util.AbstractFileClassLoader
import scala.reflect.io.VirtualDirectory
import scala.reflect.runtime.ReflectionUtils

class ReflectionUtilsTest {
  private val classLoader: ClassLoader = getClass.getClassLoader

  @Test
  def loadsStaticSingletonModuleByName(): Unit = {
    val singleton: AnyRef = ReflectionUtils.staticSingletonInstance(
      classLoader,
      "org_scala_lang.scala_reflect.ReflectionUtilsStaticSingletonFixture"
    )

    assertThat(singleton).isSameAs(ReflectionUtilsStaticSingletonFixture)
    assertThat(ReflectionUtilsStaticSingletonFixture.message).isEqualTo("static-singleton")
  }

  @Test
  def invokesInnerSingletonAccessorOnOuterInstance(): Unit = {
    val outer: ReflectionUtilsOuterFixture = new ReflectionUtilsOuterFixture("outer-id")

    val singleton: AnyRef = ReflectionUtils.innerSingletonInstance(outer, "NestedSingleton")

    assertThat(singleton).isSameAs(outer.NestedSingleton)
    assertThat(outer.NestedSingleton.description).isEqualTo("inner-singleton:outer-id")
  }

  @Test
  def describesAbstractFileClassLoaderClasspath(): Unit = {
    val root: VirtualDirectory = new VirtualDirectory("reflection-utils-root", None)
    val loader: AbstractFileClassLoader = new AbstractFileClassLoader(root, classLoader)

    val description: String = ReflectionUtils.show(loader)

    assertThat(description).contains("AbstractFileClassLoader")
    assertThat(description).contains("reflection-utils-root")
  }
}

object ReflectionUtilsStaticSingletonFixture {
  def message: String = "static-singleton"
}

class ReflectionUtilsOuterFixture(private val id: String) {
  object NestedSingleton {
    def description: String = s"inner-singleton:$id"
  }
}

File: tests/src/plexus/plexus-utils/1.0-alpha-3/src/test/java/plexus/plexus_utils/ReflectionUtilsTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package plexus.plexus_utils;

import java.lang.reflect.Field;

import org.codehaus.plexus.util.ReflectionUtils;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class ReflectionUtilsTest {
    @Test
    void findsFieldsDeclaredOnSuperclasses() {
        Field field = ReflectionUtils.getFieldByNameIncludingSuperclasses("inheritedValue", ChildBean.class);

        assertThat(field).isNotNull();
        assertThat(field.getName()).isEqualTo("inheritedValue");
        assertThat(field.getDeclaringClass()).isEqualTo(ParentBean.class);
    }

    @Test
    void findsFieldsDeclaredOnTargetClass() {
        Field field = ReflectionUtils.getFieldByNameIncludingSuperclasses("name", ChildBean.class);

        assertThat(field).isNotNull();
        assertThat(field.getName()).isEqualTo("name");
        assertThat(field.getDeclaringClass()).isEqualTo(ChildBean.class);
        assertThat(field.getType()).isEqualTo(String.class);
    }

    public static class ParentBean {
        public String inheritedValue;
    }

    public static class ChildBean extends ParentBean {
        public String name;
    }
}
